File: baekjoon/StringQuestion/077-4659-비밀번호발음하기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public boolean checkVowel(char c) throws IOException {
        char[] vowel = {'a', 'e', 'i', 'o', 'u'};
        for (int i = 0; i < vowel.length; i++) {
            if (c == vowel[i]) return true;
        }
        return false;
    }

    public boolean pwdCheck(String pwd) throws IOException {
        boolean hasVowel = false, isPrevVowel;
        char prev = pwd.charAt(0), curr = 0;
        int equalTypeCnt = 0;

        if (checkVowel(prev)) {
            hasVowel = true;
            isPrevVowel = true;
            equalTypeCnt = 1;
        }
        else {
            isPrevVowel = false;
            equalTypeCnt = -1;
        }

        for (int i = 1; i < pwd.length(); i++) {
            curr = pwd.charAt(i);
            
            if (checkVowel(curr)) {
                hasVowel = true;
                if (isPrevVowel) equalTypeCnt++;
                else equalTypeCnt = 1;
                isPrevVowel = true;
            }
            else {
                if (isPrevVowel) equalTypeCnt = -1;
                else equalTypeCnt--;
                isPrevVowel = false;
            }

            if (curr == prev && curr != 'e' && curr != 'o') {
                return false;
            }

            if (equalTypeCnt == 3 || equalTypeCnt == -3) {
                return false;
            }

            prev = curr;
        }

        if (hasVowel) return true;
        else return false;
    }
    
    public void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String input;
        while (true) {
            input = br.readLine();
            if (input.equals("end")) break;
            if (pwdCheck(input)) System.out.println("<"+input+"> is acceptable.");
            else System.out.println("<"+input+"> is not acceptable.");
        }
        br.close();
    }

    public static void main(String[] args) throws IOException {
        new Main().solution();
    }
}

File: baekjoon/StringQuestion/077-4659-비밀번호발음하기.py
import sys; input = sys.stdin.readline


def check_pwd(pwd: str) -> bool:
    hasVowel = False
    vowel = set(["a", "e", "i", "o", "u"])
    prev = pwd[0]
    is_prev_vowel = False
    equal_cnt = 0
    if prev in vowel:
        hasVowel = True
        is_prev_vowel = True
        equal_cnt = 1
    else:
        is_prev_vowel = False
        equal_cnt = -1

    for i in range(1, len(pwd)):
        c = pwd[i]
        if c in vowel:
            hasVowel = True
            if is_prev_vowel: equal_cnt += 1
            else: equal_cnt = 1
            is_prev_vowel = True
        else:
            if is_prev_vowel: equal_cnt = -1
            else: equal_cnt -= 1
            is_prev_vowel = False

        if prev == c and c != "e" and c != "o":
            return False

        if equal_cnt == 3 or equal_cnt == -3:
            return False
        
        prev = c

    if (hasVowel): return True
    else: return False


while (line := input().rstrip()) != "end":
    if check_pwd(line):
        print(f"<{line}> is acceptable.")
    else:
        print(f"<{line}> is not acceptable.")
